src/main/java/io/github/thebusybiscuit/slimefun4/implementation/guide/SurvivalSlimefunGuide.java

```diff
@@ -289,7 +289,7 @@ private void displaySlimefunItem(ChestMenu menu, ItemGroup itemGroup, Player p,
             menu.addItem(index, new CustomItemStack(ChestMenuUtils.getNoPermissionItem(), sfitem.getItemName(), message.toArray(new String[0])));
             menu.addMenuClickHandler(index, ChestMenuUtils.getEmptyClickHandler());
         } else if (isSurvivalMode() && research != null && !profile.hasUnlocked(research)) {
-            menu.addItem(index, new CustomItemStack(ChestMenuUtils.getNotResearchedItem(), ChatColor.WHITE + ItemUtils.getItemName(sfitem.getItem()), "&4&l" + Slimefun.getLocalization().getMessage(p, "guide.locked"), "", "&a> Click to unlock", "", "&7Cost: &b" + research.getCost() + " Level(s)"));
+            menu.addItem(index, new CustomItemStack(ChestMenuUtils.getNotResearchedItem(), ChatColor.WHITE + ItemUtils.getItemName(sfitem.getItem()), "&4&l" + Slimefun.getLocalization().getMessage(p, "guide.locked"), "", "&a>" + Slimefun.getLocalization().getMessage(p, "guide.click-unlock"), "", "&7" + Slimefun.getLocalization().getMessage(p, "guide.Cost") + research.getCost() + Slimefun.getLocalization().getMessage(p, "guide.Level")));
```

Comment:
the message key should be lowercase and use hyphens (`-`) to connect words.
And I don't think string concatenation is good here, better use placeholder.

Reply:
No i should not be Lowercase it works with this setup!  :)

Reply:
The key should be lowercase, or it will break the naming convention.
Also, consider using a placeholder instead of concatenating 3 parts for the cost line.

Reply:
Hmmm did you mean "getMessage" or "getMessage(p, "guide.Cost")"
Should it be named getMessage(p, "guide.cost") ?

Reply:
Yes it should be "guide.cost" the whole key should be lowercase.

As for the formatting and placeholders look at the javadocs or some examples for the String#format method to find how it works.

Reply:
I don't think I need to read about it because the original code, i.e. the red line, doesn't use a placeholder either. It's actually a function query called research.getCost() its not a placeholder so please explain to me exactly what you mean by placeholders?